recipe/meta.yaml

```diff
@@ -25,24 +46,37 @@ requirements:
     - numpy
     - cython >=0.19
     - cftime >=1.0.1
+    # need to list hdf5 and libnetcdf twice to get version pinning from conda_build_config
+    # and build pinning from {{ mpi_prefix }}
     - hdf5
+    - hdf5 * {{ mpi_prefix }}_*
     - libnetcdf
+    - libnetcdf * {{ mpi_prefix }}_*
+    - {{ mpi }}  # [mpi != 'nompi']
+    - mpi4py  # [mpi != 'nompi']
   run:
     - python
     - setuptools
     - {{ pin_compatible('numpy') }}
     - cftime
-    - hdf5
-    - libnetcdf
+    - hdf5 * {{ mpi_prefix }}_*
+    - libnetcdf * {{ mpi_prefix }}_*
+    - {{ mpi }}  # [mpi != 'nompi']
+    - mpi4py  # [mpi != 'nompi']
+    - openssh  # [mpi == 'openmpi']
 
 test:
+  files:
+    - parallel_test.py  # [mpi != 'nompi']
   imports:
     - netCDF4
     - cftime
   commands:
     - ncinfo -h
     - nc4tonc3 -h
     - nc3tonc4 -h
+    - mpirun -np 4 python parallel_test.py  # [mpi == 'mpich']
```

Comment:
I Interesting the `4` just worked here. I thought you would need to use ``{{ CPU_COUNT }}`` to make it work properly on the CIs.